src/shared/watchedFiles.ts

```diff
@@ -36,7 +36,7 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
      * Load in filesystem items, doing any parsing/validaton as required. If it fails, throws
      * @param path A string with the absolute path to the detected file
      */
-    protected abstract load(path: string): Promise<T>
+    protected abstract load(path: string): Promise<T | undefined>
```

Comment:
Hmm, this was done defensively since this would open the door to looking at all YAML files...I was thinking `validateTemplate` would error out. Looks like that's not the case, looks like general YAML parsing specifically has to break for an error to be thrown (e.g. save a template.yaml containing `{ASD}ASD{asd}ASD:asd`). Might be worth looking into  retooling the `CloudFormation.load` function

Reply:
I think returning `undefined` makes sense for communicating that the path could not be loaded. Throwing errors for flow control isn't great since that knowledge is only found in the docstring. So now I'm wondering if the responsibility for error handling should be left to the subclasses (like you have in this PR). It is potentially more repetitive but also more granular.